### PR TITLE
CLN: move pytest config to pyproject.toml

### PR DESCRIPTION
**Pull request by fangchenli**
According to https://docs.pytest.org/en/stable/customize.html, using `setup.cfg`as pytest config file is not recommended. So moved everything to `pyproject.toml`

### CODE REVIEW

**Comment by jbrockmendel** [comment]
I'm reticent to do this.  The everything-in-one-file nature of setup.cfg is really nice improvement over the older proliferation of config files

**Comment by twoertwein** [comment]
> I'm reticent to do this. The everything-in-one-file nature of setup.cfg is really nice improvement over the older proliferation of config files

Isn't pyproject.toml the "new" everything-in-one file? I would love to be able to use `poetry update` in the source tree to get all dependencies installed/updated (would need to move all the depdencies to pyproject.toml).

**Comment by jreback** [comment]
can we move *everything* to pyproject.toml? (or is this not supported for some things yet)?

**Comment by fangchenli** [comment]
> can we move _everything_ to pyproject.toml? (or is this not supported for some things yet)?

We can also move `isort` to pyproject.toml. But other tools don't support it yet.

**Comment by fangchenli** [comment]
And seems like mypy is going to support pyproject.toml soon https://github.com/python/mypy/pull/10219.

**Comment by MarcoGorelli** [comment]
Black doesn't support setup.cfg, flake8 doesn't support pyproject.toml, and the xkcd comic about standards is as relevant as ever 😆 

Anyway if the pytest docs advise against setup.cfg for pytest, then this PR seems like a good idea, thanks for doing it

**Comment by fangchenli** [comment]
LOL... Let me put it here.

**Review by MarcoGorelli — APPROVED** [review]
Looks good to me.

For anyone wondering where `-s` has gone, the [pytest docs](https://docs.pytest.org/en/6.2.x/reference.html#command-line-flags) say:

> -s                    shortcut for --capture=no.

File: ci/deps/actions-37-minimum_versions.yaml [hunk 1]
```diff
@@ -6,7 +6,7 @@ dependencies:
 
   # tools
   - cython=0.29.21
-  - pytest=5.0.1
+  - pytest=6.0
```
Comment (jreback) [inline ci/deps/actions-37-minimum_versions.yaml:9]: can we do >=

**Comment by jreback** [comment]
this looks like a net improvement to me, sure we can't move everything yet, but newer versions you can (e.g. isort)

cc @jorisvandenbossche @jbrockmendel

**Comment by jbrockmendel** [comment]
as long as we have someone who is competent with toml to maintain this file, no objection here

**Review by jreback — CHANGES_REQUESTED** [review]

**Comment by jreback** [comment]
hmm somehow a conflict, can you merge master

**Comment by MarcoGorelli** [comment]
Thanks @fangchenli !

**Comment by simonjayhawkins** [comment]
@fangchenli running the tests locally, test-data.xml is generated and is an untracked file. probably need to add to git ignore to clean up.

**Pull request merged**